weather_mv/loader_pipeline/bq.py

```diff
@@ -154,6 +158,20 @@ def __post_init__(self):
         with open_dataset(self.first_uri, self.xarray_open_dataset_kwargs,
                           self.disable_grib_schema_normalization, self.tif_metadata_for_datetime,
                           is_zarr=self.zarr) as open_ds:
+
+            # Find the grid_resolution. In case of single point we can't find grid resolution.
+            if open_ds['latitude'].size > 1 and open_ds['longitude'].size > 1:
+                # consider that Grid is regular.
```

Comment:
That the latitudes and longitudes are larger than 1 doesn't seem like that rigorous of a test that the grid is regular.

Reply:
Maybe the comment and log message is wrong?

Reply:
Hey @alxmrs, I wanted to discuss the different possibilities in our pipeline when the user provides a flag(`create_polygon`) but the input file contains grid points as a **single point**. Here are the options we have:

1. We can display a warning message to the user, ingest `None` in the Polygon, and allow the pipeline to continue running smoothly.
2. Alternatively, we can give an error to the user at this stage and terminate the pipeline.

Currently, I have implemented the `first option` in the code. I would appreciate your input on whether this approach is suitable, or if we should consider moving towards the `second option`.

Reply:
As discussed, let's do option (1). However, to address my original comment: Let's update this comment. "Regular" is an overloaded term: it can both mean "more than one grid point" (in this context) and "on an even latitude longitude grid" (what I've seen "regular" typically means).